src/Parallel/Callback.hpp

```diff
@@ -30,6 +60,16 @@ class Callback : public PUP::able {
   explicit Callback(CkMigrateMessage* msg) : PUP::able(msg) {}
   virtual void invoke() = 0;
   virtual void register_with_charm() = 0;
+  /*!
+   * \brief Returns if this callback is equal to the one passed in.
+   *
+   * \details If the `std::optional` doesn't have a value, then the pointer
+   * cast between the two callbacks failed. If it does have a value, then the
```

Comment:
There's some sort of context missing for this documentation.  There doesn't seem to be any mention of pointers or casts anywhere else.